src/Storages/VirtualColumnUtils.cpp

```diff
@@ -138,6 +142,35 @@ NameSet getVirtualNamesForFileLikeStorage()
     return getCommonVirtualsForFileLikeStorage().getNameSet();
 }
 
+std::string_view findHivePartitioningInPath(const String & path)
```

Comment:
This function won't work for non contiguous partitioning, but I don't think such thing exists